custom_metrics/markup.js

```diff
@@ -0,0 +1,647 @@
+
+        var metadata = document.querySelector('meta[name=generator][content^="AMP Plugin"]');
+        if (metadata) {
+          result.amp_plugin =  metadata.getAttribute('content') ?? null;
+        }
+
+        return result;
+      }
+      catch(e) {
+        return logError("amp", e);
+      }
+    })(),
+
+    // data-nosnippet use 
+    // Used by: SEO
+    'data_nosnippet': (() => { 
+      try {    
+        // https://support.google.com/webmasters/answer/79812?hl=en
+        // https://developers.google.com/search/reference/robots_meta_tag
+        var validNodes = document.querySelectorAll('span[data-nosnippet], div[data-nosnippet], section[data-nosnippet]');
+        var allNodes = document.querySelectorAll('[data-nosnippet]');
+        return { valid: validNodes.length, wrong_tag_type: allNodes.length - validNodes.length};
+      }
+      catch(e) {
+        return logError("data_nosnippet", e);
+      }
+    })(),
+
+    // markup info 
+    // Used by: Markup
+    'deprecated_elements': (() => {  
+      try {   
+        // https://html.spec.whatwg.org/multipage/obsolete.html#non-conforming-features
+        // Array.from(document.querySelectorAll('dfn[data-dfn-type] code')).map(e => e.innerText).join(',')
+        let result = {};
+        let deprecatedNodes = [...document.querySelectorAll('applet,acronym,bgsound,dir,noframes,isindex,keygen,listing,menuitem,nextid,noembed,plaintext,rb,rtc,strike,xmp,basefont,big,blink,center,font,multicol,nobr,spacer,tt,frameset,frame')];
+
+        deprecatedNodes.forEach((n) => {
+          let t = n.tagName.toLowerCase();
+            if (result[t])
+              result[t]++;
+            else
+              result[t] = 1;
+        });
+
+        return result;
+      }
+      catch(e) {
+        return logError("deprecated_elements", e);
+      }
+    })(),
+
+    // svg use
+    // Used by Markup
+    'svgs': (() => {   
+      try {   
+        let result = {};
+
+        result.svg_element_total = document.querySelectorAll('svg').length;
+        result.svg_img_total = document.querySelectorAll('img[src*=".svg"]').length;
+        result.svg_object_total = document.querySelectorAll('object[data*=".svg"]').length;
+        result.svg_embed_total = document.querySelectorAll('embed[src*=".svg"]').length;
+        result.svg_iframe_total = document.querySelectorAll('iframe[src*=".svg"]').length;
+
+        result.svg_total = result.svg_element_total+result.svg_img_total+result.svg_object_total+result.svg_embed_total+result.svg_iframe_total;
+
+        return result;
+      }
+      catch(e) {
+        return logError("svgs", e);
+      }
+    })(),
+
+    // app
+    // Used by Markup
+    'app': (() => {   
+      try {   
+        let result = {};
+
+        //  `<div id="app">` 
+        result.app_id_present = !!document.getElementById("app");
+
+        // `<meta name="theme-color">` 
+        result.meta_theme_color_present = !!document.querySelector('meta[name="theme-color"]');
```

Comment:
May be interesting to return the actual theme color value or null if not present.

Reply:
This was a specific request from Markup. It's one way they determine if the page is an app. I'll update it to return null or the color.